app/models/payment.rb

```diff
@@ -189,7 +189,56 @@ def sync_with_payout_processor
     sync_with_paypal if processor == PayoutProcessorType::PAYPAL
   end
 
+  def as_json(options = {})
+    json = {
+      id: external_id,
+      amount: format("%.2f", (amount_cents || 0) / 100.0),
+      currency: currency,
+      status: state,
+      created_at: created_at,
+      processed_at: state == COMPLETED ? updated_at : nil,
+      payment_processor: processor,
+      bank_account_visual: bank_account&.account_number_visual,
+      paypal_email: payment_address
+    }
+
+    if options[:include_sales]
+      json[:sales] = successful_sales.map { |sale| sale.as_json(version: 2) }
+      json[:refunded_sales] = refunded_sales.map { |sale| sale.as_json(version: 2) }
+      json[:disputed_sales] = disputed_sales.map { |sale| sale.as_json(version: 2) }
+    end
+
+    json
+  end
+
+  def successful_sales
+    Purchase.where(purchase_success_balance_id: balance_ids)
+            .not_chargedback_or_chargedback_reversed
+            .not_fully_refunded
+            .includes(:link)
+            .distinct
+            .order(created_at: :desc, id: :desc)
```

Comment:
```suggestion
    Purchase.where(purchase_success_balance_id: balance_ids)
            .includes(:link)
            .distinct
            .order(created_at: :desc, id: :desc)
```
All purchases matching the `purchase_success_balance_id` should be included here I think, refunded and disputed sales will be listed separately under the respective heads. Adding `not_fully_refunded` and `not_chargedback_or_chargedback_reversed` scopes is not needed, as it will exclude sales that got refunded/disputed in some other payout as well i.e. `purchase_refund_balance_id` / `purchase_chargeback_balance_id` is not in this payout's `balance_ids`.